validator/validator/validator.py

```diff
@@ -276,24 +282,19 @@ def sync(self):
 
     def set_weights(self):
         if len(self.hotkeys) != len(self.metagraph.hotkeys):
-            # resize
-            new_scores = [0.0] * self.metagraph.n.item()
-
-            length = len(self.hotkeys)
-            new_scores[:length] = self.scores[:length]
-
-            self.scores = new_scores
+            self.metrics.resize(len(self.hotkeys))
```

Comment:
You can get the size of the hotkeys from the metagraph directly. No need to pass it in.